features/frameworks/COMPONENT_FPGA_CI_TEST_SHIELD/include/fpga_ci_test_shield/test_utils.h

```diff
@@ -215,9 +217,11 @@ template<typename PortType, typename FunctionType, FunctionType f>
 void test_peripheral(PortType &port)
 {
     if (port.empty()) {
-        utest_printf("%d - Could not find pins to test peripheral\n", port.peripheral);
+        if (port.peripheral != NC) {
```

Comment:
shouldn't port.empty() already do the trick do we need additional if here? If yes, lets move it to line 219 (condtion && condition)

Reply:
No, as the following else is valid only for the first condition